Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth
# Uldum Rework

## Developer Changelog:
### Gameplay:
- Added a brand new crisis: Amathet Invasion.
- Added Suntouched heresy for the Titanic religion. Amathet and Lost City (aka Asova) provinces are now Suntouched, along with the local Sethir dynasty.
- Added Uldum Cloaker event, which should fire on decade pulse starting from year 600, optimally should fire near 600 in first bookmark and 610 in latter two. This option is also regulated by a gamerule and can be disabled.
- Added events for the Dark Pharaoh (Tekahn in 600+ startdates who starts as a shadow priest level 4) that can fire if a character owns the county of Neferset, is a Tol'vir and a Shadow priest and Uldum is open to everyone.
- Added Titanic branches for Suntouched religion + allowed Priest (Holy Light) class for their followers.
- Added two Tol'vir mercenary companies: Tah-Ankhat and Tah-Bosari.

### Landed titles:
- Renamed duchy and county of Tol'nariv into Amathet. County of Dulized is now known as Nahkotep.
- Split Duchy of Asova from Neferset duchy, they start off as Neferset Suntouched vassals ruled by General Husam in 583+ startdates.
- Expanded title history for Empire-tier Uldum and Kingdom of Orsis.
- County of Ammon now starts as being owned by duchy of Neferset in all startdates.
- County of Phetessin is now a tribal county, controlled by pygmies/ogres.

### Character history:
- Lore characters: Added General Husam, Prophet Barim, Prophet Hadassi and General Ammantep. Berwebot of Orsis is now known correctly as Itesh.

### Other:
- Tol'vir kings are now referred to as Pharaohs.
- Extended Voidfarer religion description.
- Added four new event pictures.
- Reworked existing Tol'vir portraits and added portraits for their stone counterparts.
- Added trait for Stone Tol'vir which grants them immortality.
- Reworded isolation divorce reason to refer to 'uncharted territories of the south' instead of the mists of Pandaria in order to apply to both isolated areas.

## Developer Notes:
- A similar mechanic to the one used in Pandaria, isolating Uldum from the rest of the world should be in place due to the presence of a cloaking mechanism shrouding the area, although the circumstances behind its malfunction are changed in GoA, since the Shattering doesn't happen in this timeline. The isolation should be optional and controlled by a gamerule.
- The Tol'vir history at the present moment is barebones, although it contains lore figures like Ninjter, Tekahn, Phaoris or Nadun (to name a few). There is a need to create some interesting characters for the region.
- Amathet Crisis in its inception serves as a counterbalance to the invading qiraji, although it can happen whenever an evil force manages to breach Uldum's defenses. They are stone tol'vir that follow a Suntouched heresy, which is much more militant compared to the orthodox Titanic beliefs.
- Additional flavor for the Neferset involve the ruler's bargain with the forces of the Void as they invade, allowing him to become a Dark Pharaoh, but a possible Voidfarer path could be also implemented. You need to be a shadow priest as that particular character to get the event (which Tekahn starts off as default in final two bookmarks).
- Replaced `is_isolated_in_pandaria_trigger`, `is_in_pandaria_or_no_mists`, `is_isolated_in_uldum_trigger`, `is_in_uldum_or_no_isolation` by `is_not_isolated_from_prev_trigger`.

## TODO:
- [x] Uldum Isolation
- [x] Tol'vir history additions
- [x] Stone Tol'vir portraits
- [x] Reworked Tol'vir portraits
- [x] Amathet Crisis
- [x] Dark Pharaoh Events
- [x] Suntouched mechanics
- [x] Loc

## How to test:
- For Dark Pharaoh events, play Pharaoh Tekahn in 603 or 605 startdates, wait until Uldum Unveiled event fires. He should receive visitors soon. You can fire this by typing `WCVOD.10100` in console.
- Amathet Crisis should fire if a major county like Amathet, Neferset, Ramkahen, Orsis or Phetessin are controlled by an evil character. You can fire this by typing `WCCRS.200` in console.
- Verify if interactions between Uldum characters and rest of the world are properly impossible until the region is unveiled. You can fire unveiling event by typing `WCULD.100` in console.
